### PR TITLE
Refine elementwiseAdd and im2col

### PR DESCRIPTION
**Pull request by chengduoZH**
fix https://github.com/PaddlePaddle/Paddle/issues/6749

### CODE REVIEW

**Comment by jacquesqiao** [comment]
这个改动实测有效果么？

**Comment by chengduoZH** [comment]
@jacquesqiao  这样改能使mnist在速度上超过了v2

**Comment by chengduoZH** [comment]
I found that [this](https://github.com/PaddlePaddle/Paddle/pull/6743/commits/7b0744edcf6d46a326b0bc7ec15cf9f1329cda4a) way can make the program run faster.

File: paddle/operators/math/im2col.cc [hunk 1]
```diff
@@ -61,14 +61,13 @@ class Im2ColFunctor<paddle::operators::math::ColFormat::kCFO,
 
     const T* im_data = im.data<T>();
     T* col_data = col->data<T>();
-
     for (int c = 0; c < channels_col; ++c) {
       int w_offset = c % filter_width;
       int h_offset = (c / filter_width) % filter_height;
-      int c_im = c / filter_width / filter_height;
+      int c_im = c / (filter_width * filter_height);
```
Comment (dzhwinter) [inline paddle/operators/math/im2col.cc]: So... Is the previous result is wrong?

File: paddle/operators/math/im2col.cc [hunk 2]
```diff
@@ -127,19 +126,26 @@ class Col2ImFunctor<paddle::operators::math::ColFormat::kCFO,
...
     const T* col_data = col.data<T>();
 
+    int w_offset = -1;
+    int h_offset = 0;
+    int c_im = 0;
     for (int c = 0; c < channels_col; ++c) {
-      int w_offset = c % filter_width;
-      int h_offset = (c / filter_width) % filter_height;
-      int c_im = c / filter_width / filter_height;
+      ++w_offset;
```
Comment (dzhwinter) [inline paddle/operators/math/im2col.cc]: Please leave a comment here. For speed.

File: paddle/operators/math/im2col.cc (hunk 1 above)
Comment (chengduoZH) [reply paddle/operators/math/im2col.cc]: No, two results are equal.

File: paddle/operators/math/im2col.cc (hunk 2 above)
Comment (chengduoZH) [reply paddle/operators/math/im2col.cc]: I found that this change has little impact on performance, and it will be reverted in the next commit.

**Pull request merged**